### PR TITLE
Minor improvements to the hierarchical model

### PR DESCRIPTION
**Pull request by rshyamsundar**
*Description of changes:*
Minor enhancements: 
* better handling of user-errors
* logging of coherency error during prediction (even if the samples are not reconciled) for better reporting of results

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

File: src/gluonts/model/deepvar_hierarchical/_estimator.py [hunk 1]
```diff
@@ -232,6 +230,15 @@ def __init__(
         use_marginal_transformation = False
         conditioning_length = 0
 
+        # This estimator doesn't work in symbolic mode.
+        if trainer.hybridize:
+            logger = logging.getLogger("gluonts").getChild("model")
```
Comment (lostella) [inline src/gluonts/model/deepvar_hierarchical/_estimator.py]: Elsewhere we follow the convention of creating a logger at the module level, named after the module, see eg https://github.com/awslabs/gluon-ts/blob/e7e06dd97029efe5f0368a39604e2572f98008a2/src/gluonts/shell/train.py#L32

File: src/gluonts/model/deepvar_hierarchical/_network.py [hunk 2]
```diff
@@ -375,12 +378,13 @@ class DeepVARHierarchicalPredictionNetwork(
...
-        reconciliation_tol: float,
         **kwargs,
     ) -> None:
         super().__init__(num_parallel_samples=num_parallel_samples, **kwargs)
         self.coherent_pred_samples = coherent_pred_samples
-        self.assert_reconciliation = assert_reconciliation
-        self.reconciliation_tol = reconciliation_tol
+        self.log_coherency_error = log_coherency_error
+        if log_coherency_error:
+            self.logger = logging.getLogger("gluonts").getChild("model")
```
Comment (lostella) [inline src/gluonts/model/deepvar_hierarchical/_network.py]: Same as above

File: src/gluonts/model/deepvar_hierarchical/_network.py [hunk 3]
```diff
@@ -297,23 +298,25 @@ def post_process_samples(self, samples: Tensor) -> Tensor:
...
-                assert (
-                    reconciliation_error(self.A, samples=coherent_samples)
-                    < self.reconciliation_tol
-                )
+            assert_shape(samples_to_return, samples.shape)
+
+        # Show coherency error: A*X_proj
+        if self.log_coherency_error:
+            coh_error = coherency_error(self.A, samples=samples_to_return)
+            self.logger.info(
```
Comment (lostella) [inline src/gluonts/model/deepvar_hierarchical/_network.py]: Same as above

File: test/model/deepvar_hierarchical/test_coherency_error.py [hunk 4]
```diff
@@ -17,7 +17,7 @@
 
 from gluonts.model.deepvar_hierarchical import (
     constraint_mat,
-    reconciliation_error,
+    coherency_error,
```
Comment (lostella) [inline test/model/deepvar_hierarchical/test_coherency_error.py:20]: Just out of curiosity: is “coherency” just a better term for the concept?

**Review by lostella — CHANGES_REQUESTED** [review]

File: test/model/deepvar_hierarchical/test_coherency_error.py (hunk 4 above)
Comment (lostella) [reply test/model/deepvar_hierarchical/test_coherency_error.py:20]: ~Isn’t it “coherence” also?~
Edit: it's fine

Comment (rshyamsundar) [reply test/model/deepvar_hierarchical/test_coherency_error.py:20]: I also prefer coherence but @lucienwerner insisted coherency is more appropriate and we used/defined that term "coherency error" in the paper.

**Review by lostella — APPROVED** [review]
Thanks @rshyamsundar! Looks good to me

**Pull request merged**